src/flb_output.c

```diff
@@ -173,6 +173,98 @@ void flb_output_coro_add(struct flb_output_instance *ins, struct flb_coro *coro)
     mk_list_add(&out_flush->_head, &ins->flush_list);
 }
 
+/*
+ * Queue a task to be flushed at a later time
+ */
+static int flb_output_task_queue_enqueue(struct flb_task_queue *queue,
+                                         struct flb_task *task,
+                                         struct flb_output_instance *out_ins,
+                                         struct flb_config *config)
+{
+    struct flb_task_enqueued *queued_task;
+
+    queued_task = flb_malloc(sizeof(struct flb_task_enqueued));
+    if (!queued_task) {
+        flb_errno();
+        return -1;
+    }
+    queued_task->out_instance = out_ins;
+    queued_task->task = task;
+    queued_task->config = config;
+
+    mk_list_add(&queued_task->_head, &queue->pending);
+    return 0;
+}
+
+/*
+ * Pop task from pending queue and flush it
+ */
+static int flb_output_task_queue_flush_one(struct flb_task_queue *queue)
+{
+    struct flb_task_enqueued *queued_task;
+    int ret;
+
+    ret = mk_list_is_empty(&queue->pending);
+    if (ret == 0) {
+        flb_error("Attempting to pop task queue with no tasks enqueued");
+        return -1;
+    }
+
+    queued_task = mk_list_entry_first(&queue->pending, struct flb_task_enqueued, _head);
+    mk_list_del(&queued_task->_head);
+    mk_list_add(&queued_task->_head, &queue->in_progress);
+    ret = flb_output_task_flush(queued_task->task,
+                                queued_task->out_instance,
+                                queued_task->config);
+
+    return ret;
+}
+
+/*
+ * Will either run or queue running a single task
+ */
+int flb_output_task_singleplex_enqueue(struct flb_task_queue *queue,
+                                       struct flb_task *task,
+                                       struct flb_output_instance *out_ins,
+                                       struct flb_config *config)
+{
+    /* Enqueue task */
+    int ret;
+    ret = flb_output_task_queue_enqueue(queue, task, out_ins, config);
+    if (ret == -1) {
+        return -1;
+    }
+
+    /* Launch task if nothing is running */
+    if (mk_list_is_empty(&out_ins->singleplex_queue->in_progress) == 0) {
+        return flb_output_task_queue_flush_one(out_ins->singleplex_queue);
+    }
+    
+    return 0;
+}
+
+/*
+ * Clear in progress task and flush a single queued task if exists
+ */
+int flb_output_task_singleplex_flush_next(struct flb_task_queue *queue)
+{
+    struct flb_task_enqueued *ended_task;
+
+    /* Remove in progress task */
+    if (!(mk_list_is_empty(&queue->in_progress) == 0)) {
```

Comment:
I think this would be more readable with a `!=`

Reply:
I personally get confused with != 0, because one would think  mk_list_is_empty would return 1 (or true) if empty, but it returns 0. the !(mk_list_is_empty...) makes it clear we are checking the list is not empty.

Reply:
I see. That makes sense. I think the most readable would be:

```
is_empty = (mk_list_is_empty(&queue->in_progress) == 0)
if (is_empty == FLB_FALSE)
```

Reply:
pending